include/deal.II/grid/reference_cell.h

```diff
@@ -2888,10 +2894,16 @@ ReferenceCell::d_linear_shape_function(const Point<dim>  &xi,
   switch (this->kind)
     {
       case ReferenceCells::Vertex:
+        return 1.0;
       case ReferenceCells::Line:
+        if constexpr (dim == 1)
+          return GeometryInfo<1>::d_linear_shape_function(xi, i);
       case ReferenceCells::Quadrilateral:
+        if constexpr (dim == 2)
+          return GeometryInfo<2>::d_linear_shape_function(xi, i);
       case ReferenceCells::Hexahedron:
-        return GeometryInfo<dim>::d_linear_shape_function(xi, i);
+        if constexpr (dim == 3)
+          return GeometryInfo<3>::d_linear_shape_function(xi, i);
```

Comment:
It's not wrong to have these `if constexpr` statements, but it's unnecessary since you already assert the correct dimension above. It's also not so easy to read because none of these `if` statements have an `else` but just fall through. Do you have plans to change these things further? Otherwise I thought that the code as it was before was perhaps easier to read.

Reply:
There's more than one way to fix this one, but the problem I ran into was that `GeometryInfo<0>::d_linear_shape_function()` doesn't exist. This wasn't a problem before because no-one called this function on a `ReferenceCells::Vertex`. Consequently we need to guard calls to `GeometryInfo<dim>::d_linear_shape_function()` to prevent link errors.

The former version is definitely easier to read - another fix would be to import this function from `GeometryInfo`.

Reply:
Yes. The long term goal is to get rid of `GeometryInfo` anyway. I leave it to you whether you already want to do that here or whether you want to leave it for later. Another option would of course have been to leave it as follows:
```
  case line:
  case quad:
  case hex:
    if constexpr (dim>0) // pointless, but avoids calling a function that doesn't exist
      call GeometryInfo
```